File: Desenvolvimento/server/src/ApplicationCore/Entities/Insumo.cs
using ApplicationCore.Entities.Identity;
using ApplicationCore.Interfaces.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    public class Insumo : IBaseEntity<long>
    {
        public long Id { get; set; }
        public string Identificador { get; set; }
        public string Nome { get; set; }
        public string Descricao { get; set; }
        public string Observacao { get; set; }
        public string Modelo { get; set; }
        public string Patrimonio { get; set; }

        public bool Ativo { get; set; }
        public DateTime DataCriacao { get; set; }
        public DateTime? DataInativacao { get; set; }

        public long? CriadoPorId { get; set; }
        public User CriadoPor { get; set; }

        public long? AlteradoPorId { get; set; }
        public virtual User AlteradoPor { get; set; }

        public long? UnidadeMedidaId { get; set; }
        public virtual UnidadeMedida UnidadeMedida { get; set; }

        public long? MarcaId { get; set; }
        public virtual Marca Marca { get; set; }

        public long? TipoInsumoId { get; set; }
        public virtual TipoInsumo TipoInsumo { get; set; }

        public long? SetorId { get; set; }
        public virtual Setor Setor { get; set; }

        public long? FornecedorId { get; set; }
        public virtual Fornecedor Fornecedor { get; set; }
    }
}

File: Desenvolvimento/client/src/@fuse/types/models/insumo.ts
import { BaseModel } from '../base-model';
import { User } from './user';
import { UnidadeMedida } from './unidade-medida';
import { Marca } from './marca';
import { Setor } from './setor';
import { TipoInsumo } from './tipo-insumo';
import { Fornecedor } from './fornecedor';

export class Insumo extends BaseModel {
    constructor(
        public id: number,
        public identificador?: string,
        public nome?: string,
        public descricao?: string,
        public observacao?: string,
        public modelo?: string,
        public patrimonio?: string,
        public ativo?: boolean,
        public dataCriacao?: Date,
        public dataInativacao?: Date,

        public criadoPorId?: number,
        public criadoPor?: User,

        public alteradoPorId?: number,
        public alteradoPor?: User,

        public unidadeMedidaId?: number,
        public unidadeMedida?: UnidadeMedida,

        public marcaId?: number,
        public marca?: Marca,

        public tipoInsumoId?: number,
        public tipoInsumo?: TipoInsumo,

        public setorId?: number,
        public setor?: Setor,

        public fornecedorId?: number,
        public fornecedor?: Fornecedor
    ) { super(); }
}
